examples/arkode/CXX_serial/ark_kpr_nestedmri.cpp

```diff
@@ -0,0 +1,1904 @@
+    w    = NV_Ith_S(y, 2);
+    uerr = SUNRabs(NV_Ith_S(yref, 0) - u);
+    verr = SUNRabs(NV_Ith_S(yref, 1) - v);
+    werr = SUNRabs(NV_Ith_S(yref, 2) - w);
+    uerrtot += uerr * uerr;
+    verrtot += verr * verr;
+    werrtot += werr * werr;
+    errtot += uerr * uerr + verr * verr + werr * werr;
+    accuracy = std::max(accuracy, uerr / SUNRabs(opts.atol +
+                                                 opts.rtol * NV_Ith_S(yref, 0)));
+    accuracy = std::max(accuracy, verr / SUNRabs(opts.atol +
+                                                 opts.rtol * NV_Ith_S(yref, 1)));
+    accuracy = std::max(accuracy, werr / SUNRabs(opts.atol +
+                                                 opts.rtol * NV_Ith_S(yref, 2)));
+    Nout++;
+
+    // Periodically output current results to screen
+    if (t >= tout)
+    {
+      tout += dTout;
+      tout = (tout > Tf) ? Tf : tout;
+      printf("  %10.6" FSYM " %10.6" FSYM " %10.6" FSYM " %10.6" FSYM
+             "   %.2" ESYM "   %.2" ESYM "   %.2" ESYM "\n",
+             t, u, v, w, uerr, verr, werr);
+    }
+  }
+  uerrtot = SUNRsqrt(uerrtot / Nt);
+  verrtot = SUNRsqrt(verrtot / Nt);
+  werrtot = SUNRsqrt(werrtot / Nt);
+  errtot  = SUNRsqrt(errtot / Nt / 3);
+  printf("   "
+         "---------------------------------------------------------------------"
+         "-------\n");
+
+  //
+  // Finalize
+  //
+
+  // Get some slow integrator statistics
+  long int nsts, natts, netfs, nfse, nfsi;
+  retval = ARKodeGetNumSteps(arkode_mem, &nsts);
+  check_flag(retval, "ARKodeGetNumSteps");
+  retval = ARKodeGetNumStepAttempts(arkode_mem, &natts);
+  check_flag(retval, "ARKodeGetNumStepAttempts");
+  retval = ARKodeGetNumErrTestFails(arkode_mem, &netfs);
+  check_flag(retval, "ARKodeGetNumErrTestFails");
+  retval = MRIStepGetNumRhsEvals(arkode_mem, &nfse, &nfsi);
+  check_flag(retval, "MRIStepGetNumRhsEvals");
+
+  // Get some intermediate integrator statistics
+  long int nstm, nattm, netfm, nfme, nfmi;
+  retval = ARKodeGetNumSteps(mid_arkode_mem, &nstm);
+  check_flag(retval, "ARKodeGetNumSteps");
+  retval = ARKodeGetNumStepAttempts(mid_arkode_mem, &nattm);
+  check_flag(retval, "ARKodeGetNumStepAttempts");
+  retval = ARKodeGetNumErrTestFails(mid_arkode_mem, &netfm);
+  check_flag(retval, "ARKodeGetNumErrTestFails");
+  retval = MRIStepGetNumRhsEvals(mid_arkode_mem, &nfme, &nfmi);
+  check_flag(retval, "MRIStepGetNumRhsEvals");
+
+  // Get some fast integrator statistics
+  long int nstf, nattf, netff, nff;
+  retval = ARKodeGetNumSteps(inner_arkode_mem, &nstf);
+  check_flag(retval, "ARKodeGetNumSteps");
+  retval = ARKodeGetNumStepAttempts(inner_arkode_mem, &nattf);
+  check_flag(retval, "ARKodeGetNumStepAttempts");
+  retval = ARKodeGetNumErrTestFails(inner_arkode_mem, &netff);
+  check_flag(retval, "ARKodeGetNumErrTestFails");
+  retval = ERKStepGetNumRhsEvals(inner_arkode_mem, &nff);
+  check_flag(retval, "ERKStepGetNumRhsEvals");
+
+  // Print some final statistics
```

Comment:
Would it be better to just use the PrintAllStats functions?

Reply:
We could call `PrintAllStats` from all three integrators, but the output would be significantly longer.  Since this example focuses on 3-time-scale adaptivity, I like that the number of time steps/attempts/rejections for each time level are printed consecutively, before we output anything about implicit solvers at each time scale.